nbdime/gitdiffdriver.py

```diff
@@ -96,19 +95,23 @@ def main(args=None):
         enable_help="enable nbdime diff driver via git config",
         disable_help="disable nbdime diff driver via git config")
 
+    return parser
+
+
+def main(args=None):
+    if args is None:
+        args = sys.argv[1:]
+
+    setup_std_streams()
+    parser = _build_arg_parser()
     opts = parser.parse_args(args)
+
     if opts.subcommand == 'diff':
         from . import nbdiffapp
-        return nbdiffapp.main(
-            [opts.a, opts.b] +
-            ['--%s' % name for name in diff_exclusives if getattr(opts, name)])
+        return nbdiffapp.main_diff(opts)
     elif opts.subcommand == 'webdiff':
-        from nbdime.webapp import nbdiffweb
-        return nbdiffweb.main(
-            [opts.a, opts.b] +
-            ['--%s' % name for name in diff_exclusives if getattr(opts, name)] +
-            [i for l in [['--%s' % arg, str(getattr(opts, arg.replace('-', '_')))] for arg in web_args] for i in l]
-            )
+        from .webapp import nbdiffweb
+        return nbdiffweb.main_diff(opts)
```

Comment:
The solution that was there was correct given the API, but not very pretty. So I refactored the API to something a little better :)